annotation/tests/test_tasks_services.py

```diff
@@ -585,3 +636,1697 @@ def test_filter_tasks_db_no_files_or_jobs(
+
+    assert objs_names_from_cache == expected_objs_names_from_cache
+    assert not_cached_usernames == expected_not_cached_usernames
+
+
+def test_get_user_names_by_user_ids():
+    user_ids = {1, 2, 3}
+    tenant = "test_tenant"
+    token = "test_token"
+
+    cached_user_names = {1: "user1", 3: "user3"}
+    not_cached_user_ids = {2}
+    user_names_from_requests = {2: "user2"}
+
+    with patch(
+        "annotation.tasks.services.get_from_cache",
+        return_value=(
+            cached_user_names,
+            not_cached_user_ids,
+        ),
+    ) as mock_get_from_cache, patch(
+        "annotation.tasks.services.get_user_names_by_request",
+        return_value=user_names_from_requests,
+    ) as mock_get_user_names_by_request, patch(
+        "annotation.tasks.services.lru_user_id_to_user_name_cache"
+    ) as mock_cache:
+
+        result = get_user_names_by_user_ids(user_ids, tenant, token)
+
+        mock_get_from_cache.assert_called_once_with(
+            objs_ids=user_ids, cache_used=mock_cache
+        )
+
+        mock_get_user_names_by_request.assert_called_once_with(
+            user_ids=list(not_cached_user_ids), tenant=tenant, token=token
+        )
+
+        mock_cache.update.assert_called_once_with(user_names_from_requests)
+
+        expected_result = {**cached_user_names, **user_names_from_requests}
+
+        assert result == expected_result
+
+
+def test_get_file_names_by_file_ids():
+    file_ids = {1, 2, 3}
+    tenant = "test_tenant"
+    token = "test_token"
+
+    cached_file_names = {1: "file1", 3: "file3"}
+    not_cached_file_ids = {2}
+    file_names_from_requests = {2: "file2"}
+
+    with patch(
+        "annotation.tasks.services.get_from_cache",
+        return_value=(
+            cached_file_names,
+            not_cached_file_ids,
+        ),
+    ) as mock_get_from_cache, patch(
+        "annotation.tasks.services.get_file_names_by_request",
+        return_value=file_names_from_requests,
+    ) as mock_get_file_names_by_request, patch(
+        "annotation.tasks.services.lru_file_id_to_file_name_cache"
+    ) as mock_cache:
+
+        result = get_file_names_by_file_ids(file_ids, tenant, token)
+
+        mock_get_from_cache.assert_called_once_with(
+            objs_ids=file_ids, cache_used=mock_cache
+        )
+
+        mock_get_file_names_by_request.assert_called_once_with(
+            file_ids=list(not_cached_file_ids), tenant=tenant, token=token
+        )
+
+        mock_cache.update.assert_called_once_with(file_names_from_requests)
+
+        expected_result = {**cached_file_names, **file_names_from_requests}
+
+        assert result == expected_result
+
+
+@pytest.fixture
+def mock_metric():
+    metric = MagicMock()
+    metric.task_from = datetime(2024, 1, 1)
+    metric.task_to = datetime(2024, 2, 1)
+    metric.agreement_metric = True
+    yield metric
+
+
+@pytest.fixture
+def mock_stats(mock_task: ManualAnnotationTask, mock_metric: Mock):
+    stat1 = MagicMock()
```

Comment:
any reason why here and in other similar fixtures you create mocks instead of creating a real object with data?

Reply:
I used mocks for this because it seemed to make more sense to mock objects as I was testing the code, and it looked nicer to me. Should I switch back to using real objects?